maven-plugins/helidon-archetype-maven-plugin/src/main/java/io/helidon/build/maven/archetype/JarMojo.java

```diff
@@ -346,4 +355,49 @@ private String coordinates(Dependency d) {
         coords += ":" + d.getVersion();
         return coords;
     }
+
+    private static class RegexValidator extends VisitorAdapter<Context> {
```

Comment:
How about moving that class to a separate file ?
Maybe move existing `Validation` to a subpackage `.config` to avoid the confusion.

And we can create `OutputValidator` which contains the current logic that uses `Validation` and add your new class `RegexValidator` ; this would make things is bit more consistent.

Reply:
What about `Validator` name because indeed `Validation` is configuration to validate generated files but `RegexValidation` validate input data ?

Reply:
I'd be tempted to call it InputValidator but it does not validate input, it truly validates the regex.

Reply:
I see what you did, you've combined these into a single class ; that's fine.